web/src/contexts/BlockContext.tsx

```diff
@@ -0,0 +1,93 @@
+import { ReactNode, createContext, useContext, useState } from 'react'
+
+import { create } from '@bufbuild/protobuf'
+
+import {
+  Block,
+  BlockKind,
+  BlockRole,
+  BlockSchema,
+  GenerateRequest,
+  GenerateRequestSchema,
+} from '../gen/es/cassie/blocks_pb'
+import { useClient as useAgentClient } from './AgentContext'
+
+type BlockContextType = {
+  blocks: Block[]
```

Comment:
How do you update blocks? Would it be helpful to have a map ID->Block so that if the backend is updating an existing block you can just update the entry in the map?

Reply:
This property should go away. I renamed `Messages` to `Blocks` to align the terminology. Instead of "pushing" blocks, the API exposed should align with the user interaction, e.g., `addMessage` (specifically for the text input), and the list of `Blocks` should be exported so that downstream components can "pull them".

I haven't thought through what this means for feeding outputs back into the APIs, but the idea was to centralize all this logic in the context. Unfortunately, I would up running out of time on Friday.